File: Elitesportweb/EliteSportsApp/screens/ClassesScreen.js
import React from 'react';
import { View, Text, StyleSheet, ScrollView } from 'react-native';

export default function ClassesScreen() {
  const classes = [
    { id: 1, name: 'CrossFit', time: '08:00 AM', instructor: 'John Doe' },
    { id: 2, name: 'Karate', time: '10:00 AM', instructor: 'Jane Smith' },
    { id: 3, name: 'Zumba', time: '06:00 PM', instructor: 'Mike Johnson' },
  ];

  return (
    <ScrollView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>My Classes</Text>
      </View>
      
      {classes.map(cls => (
        <View key={cls.id} style={styles.classCard}>
          <Text style={styles.className}>{cls.name}</Text>
          <Text style={styles.classTime}>{cls.time}</Text>
          <Text style={styles.classInstructor}>Instructor: {cls.instructor}</Text>
        </View>
      ))}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  header: {
    padding: 20,
    backgroundColor: 'white',
    marginBottom: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#333',
  },
  classCard: {
    backgroundColor: 'white',
    margin: 10,
    padding: 20,
    borderRadius: 10,
  },
  className: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#f36100',
    marginBottom: 5,
  },
  classTime: {
    fontSize: 16,
    color: '#333',
    marginBottom: 5,
  },
  classInstructor: {
    fontSize: 14,
    color: '#666',
  },
});

File: Elitesportweb/elite_sports_flutter/lib/screens/classes_screen.dart
import 'package:flutter/material.dart';
import '../models/class_model.dart';

class ClassesScreen extends StatelessWidget {
  final List<ClassModel> classes;

  const ClassesScreen({super.key, required this.classes});

  Widget _buildClassCard(ClassModel classModel) {
    return Container(
      margin: const EdgeInsets.only(bottom: 16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(16),
        boxShadow: [
          BoxShadow(
            color: Colors.grey.withOpacity(0.1),
            blurRadius: 8,
            offset: const Offset(0, 4),
          ),
        ],
      ),
      child: Column(
        children: [
          Container(
            width: double.infinity,
            padding: const EdgeInsets.all(20),
            decoration: BoxDecoration(
              gradient: LinearGradient(
                colors: _getClassColors(classModel.type),
                begin: Alignment.topLeft,
                end: Alignment.bottomRight,
              ),
              borderRadius: const BorderRadius.only(
                topLeft: Radius.circular(16),
                topRight: Radius.circular(16),
              ),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  children: [
                    Icon(
                      _getClassIcon(classModel.type),
                      color: Colors.white,
                      size: 24,
                    ),
                    const SizedBox(width: 12),
                    Expanded(
                      child: Text(
                        classModel.name,
                        style: const TextStyle(
                          color: Colors.white,
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 8),
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
                  decoration: BoxDecoration(
                    color: Colors.white.withOpacity(0.2),
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Text(
                    classModel.type.toUpperCase(),
                    style: const TextStyle(
                      color: Colors.white,
                      fontSize: 12,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ],
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(20),
            child: Column(
              children: [
                _buildInfoRow(Icons.access_time, 'Time', classModel.time),
                const SizedBox(height: 12),
                _buildInfoRow(Icons.timer, 'Duration', classModel.duration),
                const SizedBox(height: 12),
                _buildInfoRow(Icons.person, 'Instructor', classModel.instructor),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildInfoRow(IconData icon, String label, String value) {
    return Row(
      children: [
        Icon(icon, color: Colors.grey.shade600, size: 20),
        const SizedBox(width: 12),
        Text(
          '$label: ',
          style: TextStyle(
            color: Colors.grey.shade600,
            fontSize: 14,
            fontWeight: FontWeight.w500,
          ),
        ),
        Expanded(
          child: Text(
            value,
            style: const TextStyle(
              fontSize: 14,
              fontWeight: FontWeight.w600,
            ),
          ),
        ),
      ],
    );
  }

  List<Color> _getClassColors(String type) {
    switch (type.toLowerCase()) {
      case 'crossfit':
        return [Colors.orange, Colors.deepOrange];
      case 'karate':
        return [Colors.red, Colors.redAccent];
      case 'zumba':
        return [Colors.purple, Colors.purpleAccent];
      default:
        return [const Color(0xFF1e3c72), const Color(0xFF2a5298)];
    }
  }

  IconData _getClassIcon(String type) {
    switch (type.toLowerCase()) {
      case 'crossfit':
        return Icons.fitness_center;
      case 'karate':
        return Icons.sports_martial_arts;
      case 'zumba':
        return Icons.music_note;
      default:
        return Icons.sports;
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey.shade50,
      appBar: AppBar(
        title: const Text('My Classes'),
        backgroundColor: const Color(0xFF1e3c72),
        foregroundColor: Colors.white,
        elevation: 0,
      ),
      body: classes.isEmpty
          ? Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Icon(
                    Icons.fitness_center,
                    size: 80,
                    color: Colors.grey.shade400,
                  ),
                  const SizedBox(height: 16),
                  Text(
                    'No Classes Assigned',
                    style: TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.bold,
                      color: Colors.grey.shade600,
                    ),
                  ),
                  const SizedBox(height: 8),
                  Text(
                    'Contact your instructor to join classes',
                    style: TextStyle(
                      fontSize: 16,
                      color: Colors.grey.shade500,
                    ),
                  ),
                ],
              ),
            )
          : ListView.builder(
              padding: const EdgeInsets.all(16),
              itemCount: classes.length,
              itemBuilder: (context, index) {
                return _buildClassCard(classes[index]);
              },
            ),
    );
  }
}